include/boost/geometry/algorithms/detail/closest_points/linear_or_areal_to_areal.hpp

```diff
@@ -0,0 +1,269 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2021, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Vissarion Fysikopoulos, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_LINEAR_OR_AREAL_TO_AREAL_HPP
+#define BOOST_GEOMETRY_ALGORITHMS_DETAIL_CLOSEST_POINTS_LINEAR_OR_AREAL_TO_AREAL_HPP
+
+#include <boost/geometry/algorithms/detail/closest_points/linear_to_linear.hpp>
+#include <boost/geometry/algorithms/detail/closest_points/utilities.hpp>
+
+#include <boost/geometry/algorithms/intersection.hpp>
+
+#include <boost/geometry/core/point_type.hpp>
+
+#include <boost/geometry/geometries/geometries.hpp>
+
+namespace boost { namespace geometry
+{
+
+#ifndef DOXYGEN_NO_DETAIL
+namespace detail { namespace closest_points
+{
+
+struct linear_to_areal
+{
+    template <typename Linear, typename Areal, typename Segment, typename Strategies>
+    static inline void apply(Linear const& linear,
+                             Areal const& areal,
+                             Segment& shortest_seg,
+                             Strategies const& strategies)
+    {
+        using most_precise_type = typename select_coordinate_type
+                <
+                    Linear,
+                    Areal
+                >::type;
+
+        using point_type = typename std::conditional
+        <
+            std::is_same<coordinate_type<Linear>, most_precise_type>::value,
```

Comment:
`geometry::coordinate_type` is not a type alias so here it should be `typename coordinate_type<Linear>::type`.

Side note. We could think about adding type aliases and variable templates for traits: `coordinate_t`, `point_t`, `dimension_v`, etc.

Reply:
> `geometry::coordinate_type` is not a type alias so here it should've been `typename coordinate_type<Linear>::type`.

sure, thanks for noticing 

> Side note. We could think about adding type aliases and variable templates for traits: `coordinate_t`, `point_t`, `dimension_v`, etc.

+1